tests/test_mpi.py

```diff
@@ -180,6 +187,57 @@ def test_custom_topology(self):
         assert f2.shape == expected[distributor.myrank]
         assert f2.size_global == f.size_global
 
+    @pytest.mark.parametrize('comm_size, topology, dist_topology', [
+        (2, (1, '*'), (1, 2)),
+        (2, ('*', '*'), (2, 1)),
+        (1, (1, '*', '*'), (1, 1, 1)),
+        (2, (1, '*', '*'), (1, 2, 1)),
+        (2, (2, '*', '*'), (2, 1, 1)),
+        (3, (1, '*', '*'), (1, 3, 1)),
+        (3, ('*', 1, '*'), (3, 1, 1)),
+        (3, ('*', '*', 1), (3, 1, 1)),
+        (4, (2, '*', '*'), (2, 2, 1)),
+        (4, ('*', 2, '*'), (2, 2, 1)),
+        (4, ('*', '*', 2), (2, 1, 2)),
+        (6, ('*', 1, '*'), (3, 1, 2)),
+        (6, ('*', '*', 1), (3, 2, 1)),
+        (6, (1, '*', '*'), (1, 3, 2)),
+        (6, ('*', '*', '*'), (3, 2, 1)),
+        (12, ('*', '*', '*'), (3, 2, 2)),
+        (12, ('*', 3, '*'), (2, 3, 2)),
+        (18, ('*', '*', '*'), (3, 3, 2)),
+        (18, ('*', '*', 9), (2, 1, 9)),
+        (18, ('*', '*', 3), (3, 2, 3)),
+        (24, ('*', '*', '*'), (6, 2, 2)),
+        (32, ('*', '*', '*'), (4, 4, 2)),
+        (8, ('*', 1, '*'), (4, 1, 2)),
+        (8, ('*', '*', 1), (4, 2, 1)),
+        (8, (1, '*', '*'), (1, 4, 2)),
+        (8, ('*', '*', '*'), (2, 2, 2)),
+        (9, ('*', '*', '*'), (3, 3, 1)),
+        (11, (1, '*', '*'), (1, 11, 1)),
+        (22, ('*', '*', '*'), (11, 2, 1)),
+        (16, ('*', '*', 1), (4, 4, 1)),
+        (16, ('*', 1, '*'), (4, 1, 4)),
+        (32, ('*', '*', 1), (8, 4, 1)),
+        (64, ('*', '*', '*'), (4, 4, 4)),
+        (64, ('*', '*', 1), (8, 8, 1)),
+        (64, ('*', 2, 1), (32, 2, 1)),
+        (64, ('*', 2, 4), (8, 2, 4)),
+        (128, ('*', '*', 1), (16, 8, 1)),
+        (231, ('*', '*', '*'), (11, 7, 3)),
+        (256, (1, '*', '*'), (1, 16, 16)),
+        (256, ('*', 1, '*'), (16, 1, 16)),
+        (256, ('*', '*', 1), (16, 16, 1)),
+        (256, ('*', '*', '*'), (8, 8, 4)),
+        (256, ('*', '*', 2), (16, 8, 2)),
+        (256, ('*', 32, 2), (4, 32, 2)),
+    ])
+    def test_custom_topology_3d_dummy(self, comm_size, topology, dist_topology):
```

Comment:
isn't this... a bit too much?

Reply:
Well, maybe a few can be dropped, but I wanted to extensively test this for some big number of ranks for the strong scaling experiments.
What is maybe redundant is the prime combs like 11, 231 etc...
Overall the above tests are very cheap (no MPI_Init, as we discussed), so I would not worry.

Reply:
thoughts @mloubout ?

Reply:
Looks a bit overkill yeah but it's very cheap so I guess it's fine

Reply:
I dropped a few as well